code/modules/jobs/job_types/chaplain/chaplain_nullrod.dm

```diff
@@ -646,3 +690,172 @@
+		successful_sneak_attack = TRUE
+
+	// Now check for if we're behind
+	var/dir_living_target_to_user = get_dir(living_target, user)
+	if(living_target.dir & REVERSE_DIR(dir_living_target_to_user))
+		successful_sneak_attack = TRUE
+
+	/// Now we'll check for things that STOP a sneak attack. Why? Because this mechanic isn't complicated enough and I must insert more ivory tower design.
+
+	if(living_target.mob_biotypes & MOB_SLIME) // SLIMES HAVE NO ANATOMY.
+		successful_sneak_attack = FALSE
+		sneak_attack_fail_message = TRUE
+
+	else if(living_target.incorporeal_move >= 1 && !HAS_TRAIT(living_target, TRAIT_REVENANT_REVEALED)) // WE CAN'T SNEAK ATTACK INCORPOREAL JERKS. BUT WE CAN SNEAK ATTACK REVEALED REVENANTS BECAUSE DUH, NULLROD.
+		successful_sneak_attack = FALSE
+		sneak_attack_fail_message = TRUE
+
+	else if(IS_HERETIC_MONSTER(living_target) && prob(50)) // IT IS HARD TO SNEAK ATTACK SOMETHING WITH TOO MANY REDUNDANT EVERYTHINGS.
+		successful_sneak_attack = FALSE
+		sneak_attack_fail_message = TRUE
+
+	else if(HAS_TRAIT(living_target, TRAIT_STABLEHEART) && prob(50)) // THEIR ANATOMY IS FUCKING WEIRD.
+		successful_sneak_attack = FALSE
+		sneak_attack_fail_message = TRUE
+
+	else if(HAS_TRAIT(living_target, TRAIT_MIND_READER) && !user.can_block_magic(MAGIC_RESISTANCE_MIND, charge_cost = 0)) // FORESIGHT SAYS 'FUCK YOU' TO SNEAK ATTACKERS. BUT IF YOU HAVE A TIN FOIL HAT, YOU'RE SAFE!
+		successful_sneak_attack = FALSE
+		sneak_attack_fail_message = TRUE
+
+	else if(user.is_blind()) // YOU CAN'T STAB PRECISELY WHAT YOU CAN'T SEE.
+		successful_sneak_attack = FALSE
+		sneak_attack_fail_message = TRUE
+
+	/// And now we'll deal with sneak attack damage modifiers.
+
+	// If our target is also unconscious for some reason, we get even more damage. Coup de grace, motherfucker.
+	if(HAS_TRAIT(living_target, TRAIT_KNOCKEDOUT))
+		sneak_attack_dice += roll("1d6")
+		new /obj/effect/temp_visual/crit(get_turf(living_target))
+
+	// If the target is rebuked, we also add some additional damage. It is the closest thing to 'studying' your target, okay?
+	if(living_target.has_status_effect(/datum/status_effect/rebuked))
+		sneak_attack_dice += 2
+
+	// If we're morbid, and the target has been dissected, we get an extra d6.
+	// The chances of this occuring are quite low, as even having this weapon means you're locked out of becoming morbid as a chaplain, but when it does come up...
+	// Or the coroner stole this blade to go hunt the recently dead...
+	if(HAS_TRAIT(user, TRAIT_MORBID) && HAS_TRAIT(living_target, TRAIT_DISSECTED))
+		sneak_attack_dice += roll("1d6")
+
+	// Baton + this weapon might be a little too much fun so we're nerfing this combination outright.
+	if(HAS_TRAIT(living_target, TRAIT_IWASBATONED))
+		sneak_attack_dice *= 0.5
+
+	/// And so we return here if we are not entitled to a sneak attack.
+	if(!successful_sneak_attack)
+		if(sneak_attack_fail_message)
+			user.balloon_alert(living_target, "sneak attack avoided!")
+		return
```

Comment:
This return can be moved up to `// And now we'll deal with sneak attack damage modifiers.` I think right? No need to calculate modifiers if it's failed

Reply:
yeah, very fair it would have failed before all these modifiers anyway